### PR TITLE
Remove test Stan models for integration tests

### PR DESCRIPTION
**Pull request by rok-cesnovar**
#### Summary

Last step of #3075

Removes all Stan test models that were used only in integration tests to check whether they compile. After #3078, we will switch to stanc3 models and these are then unnecessary. There are 28 remaining Stan files in here, which are used for unit tests.

This is built on top of https://github.com/stan-dev/stan/pull/3078 which should be merged first

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company):
Rok Češnovar

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
- Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

### CODE REVIEW

**Comment by bob-carpenter** [comment]
@rok-cesnovar:  Are the tests just being moved from old parser tests to stanc3 tests?  I think we still want to test that all these models actually generate code that can be compiled.  This is getting even more critical now with all the action on the template metaprogramming and complex/function type side.

**Comment by rok-cesnovar** [comment]
Yes, see https://github.com/stan-dev/stan/pull/3078 and https://github.com/stan-dev/stan/issues/3075

Tldr: Test from stanc3 repo are used for integration tests. Models from the example-models repo will also be tested now. So this is an expansion of test coverage.

**Comment by rok-cesnovar** [comment]
The main reason for this is that tests with new feature, signatures, etc only get added to the stanc3 repo and it makes no sense to duplicate them here.

**Comment by rok-cesnovar** [comment]
#3078 was merged so this is now ready for review. It is deleting 861 files, but you can see in the files changed view that it is only removing .stan and .R files (those accompany some .stan files).

**Review by WardBrian — APPROVED** [review]
Confirmed it is only removing .stan and .R files, looks good to me
